docs/apache-airflow/best-practices.rst

```diff
@@ -619,3 +621,219 @@ Prune data before upgrading
 ---------------------------
 
 Some database migrations can be time-consuming.  If your metadata database is very large, consider pruning some of the old data with the :ref:`db clean<cli-db-clean>` command prior to performing the upgrade.  *Use with caution.*
+
```

Comment:
Very fantastic and detailed doc Jarek! :tada: 

If read through entirely it should give folks who are new to Airflow a good summary of the different task execution options but also they will learn more about how the Airflow execution environment itself works (where dependencies are sourced from, how tasks can conflict on shared execution hosts, etc).

It looks like a lot of comments below, but there was a lot of text to read through :) Most of the comments are just grammar/naming.

Reply:
Thanks @o-nikolas :). English Grammar is not my strongest skill :). We don't have articles in Polish at all, that's why I almost never know when to use a/the :). I really appreciate you taking time to look at it and correct it!

Reply:
No worries at all! I only speak one language so I can't judge anyone else :smile:

Reply:
I added one more sentence to clarify (and rebased too).